File: word_count.py
from pyspark import SparkContext, SparkConf
import os
import shutil

if __name__ == "__main__":
    conf = SparkConf().setAppName("appName").setMaster("local")
    sc = SparkContext(conf=conf)
    sourceDataRDD = sc.textFile("file:////Users/lvshuo/Desktop/tmp/hello.txt")
    wordsRDD = sourceDataRDD.flatMap(lambda line: line.split())
    keyValueWordsRDD = wordsRDD.map(lambda s: (s, 1))
    wordCountRDD = keyValueWordsRDD.reduceByKey(lambda a, b: a + b)

    outputPath = "/Users/lvshuo/Desktop/tmp/wordcount"
    if os.path.exists(outputPath):
        shutil.rmtree(outputPath)
    wordsRDD.saveAsTextFile("file://" + outputPath)
    print(wordCountRDD.collect())

File: spark-rdd-basic/src/main/scala/com/claus/spark/core/wordcount/WordCount.scala
package com.claus.spark.core.wordcount

import org.apache.spark.rdd.RDD
import org.apache.spark.{SparkConf, SparkContext}

object WordCount {

  def main(args: Array[String]): Unit = {

    // prepare spark environment
    val sparkConf = new SparkConf().setMaster("local").setAppName("WordCountDemo")
    val sc = new SparkContext(sparkConf)

    // transform
    val fileRDD: RDD[String] = sc.textFile("input")

    val wordRDD: RDD[String] = fileRDD.flatMap(_.split(" "))

    val word2OneRDD: RDD[(String, Int)] = wordRDD.map((_, 1))

    val wordCountRDD: RDD[(String, Int)] = word2OneRDD.reduceByKey(_ + _)

    // action
    val wordCount: Array[(String, Int)] = wordCountRDD.collect()
    wordCount.foreach(println)

    sc.stop()
  }

}
